# Práctica 1.1 — Probabilidad y Estadística

## Análisis de la evolución poblacional mundial

## Objetivo

Utilizar un conjunto de datos real para comprobar el funcionamiento del entorno de análisis de datos y aplicar medidas estadísticas descriptivas básicas utilizando Pandas y NumPy.

## Actividad 1 — Identificación de la fuente

**Propósito:** reconocer la procedencia del conjunto de datos antes de utilizarlo y documentar correctamente su fuente.

Los datos utilizados en esta práctica provienen de **Our World in Data (OWID)**. La fuente original es **United Nations, World Population Prospects (2024)**.

**Cita:**

> UN, World Population Prospects (2024) – processed by Our World in Data. “Annual change in population – UN WPP” [dataset]. United Nations, “World Population Prospects”; United Nations, “World Population Prospects - Interim Update” [original data].

### Preguntas

1. **¿Quién produjo originalmente los datos?** United Nations.
2. **¿Qué organización los procesa para su utilización en Our World in Data?** Our World in Data.
3. **¿Qué representa la variable que analizarás?** Representa el cambio anual de la población, es decir, cuánto aumenta o disminuye el número de habitantes de una entidad respecto al año anterior.

## Actividad 2 — Carga y exploración de datos

**Propósito:** comprobar que el entorno de Python, Pandas y Jupyter funciona correctamente y realizar una primera exploración del dataset.

### 2.1 Importar las bibliotecas

In [1]:
from pathlib import Path
import pandas as pd
import numpy as np

### 2.2 Cargar el archivo CSV

Se utiliza una ruta relativa para que el notebook funcione tanto desde la raíz del proyecto como desde la carpeta `notebooks`.

In [2]:
raiz_proyecto = Path.cwd() if (Path.cwd() / "data").exists() else Path.cwd().parent
ruta_csv = raiz_proyecto / "data" / "raw" / "annual-population-growth" / "annual-population-growth.csv"
df = pd.read_csv(ruta_csv)
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


### 2.3 Explorar el DataFrame

Se revisan las primeras filas, dimensiones, columnas, tipos de datos, estadísticas descriptivas, valores faltantes y cantidad de entidades.

In [3]:
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [4]:
print("Dimensiones:", df.shape)
print("Nombres de columnas:", df.columns.tolist())
print("\nTipos de datos:")
display(df.dtypes)

Dimensiones: (38400, 5)
Nombres de columnas: ['Entity', 'Code', 'Year', 'Annual change in population', 'Annual population change (Projected)']

Tipos de datos:


Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [5]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [6]:
print("Valores faltantes por columna:")
display(df.isnull().sum())
print("Cantidad de entidades diferentes:", df["Entity"].nunique())

Valores faltantes por columna:


Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

Cantidad de entidades diferentes: 256


### Respuestas de la Actividad 2

1. **¿Cuántas filas y columnas tiene el dataset?** Tiene **38,400 filas y 5 columnas**.
2. **¿Qué tipo de datos contiene la variable `Year`?** Es de tipo entero (`int64`).
3. **¿Existen valores faltantes?** Sí. `Code` tiene 1,200; `Annual change in population`, 19,712; y `Annual population change (Projected)`, 18,688. `Entity` y `Year` no tienen valores faltantes. Los faltantes de las dos variables poblacionales se deben a que una contiene datos históricos y la otra proyecciones.
4. **¿Cuántas entidades diferentes contiene el dataset?** Contiene **256 entidades**.

## Actividad 3 — Estadística descriptiva

**Propósito:** aplicar medidas estadísticas básicas a una variable poblacional.

Se selecciona el año histórico **2020** y se analiza la variable `Annual change in population`. Se eliminan únicamente los valores faltantes de esta variable antes de calcular los estadísticos.

In [7]:
variable = "Annual change in population"
df_2020 = df.loc[(df["Year"] == 2020) & df[variable].notna()].copy()
print("Registros históricos de 2020:", len(df_2020))
df_2020[["Entity", "Year", variable]].head()

Registros históricos de 2020: 256


,Entity,Year,Annual change in population
69,Afghanistan,2020,1212852.0
219,Africa (UN),2020,32815675.0
369,Albania,2020,-13061.0
519,Algeria,2020,747543.0
669,American Samoa,2020,-451.0


In [8]:
cambio_2020 = df_2020[variable]

estadisticos_2020 = pd.Series({
    "Media": cambio_2020.mean(),
    "Mediana": cambio_2020.median(),
    "Varianza": cambio_2020.var(),
    "Desviación estándar": cambio_2020.std(),
    "Mínimo": cambio_2020.min(),
    "Máximo": cambio_2020.max(),
}, name="Cambio anual de población en 2020")

estadisticos_2020.to_frame().style.format("{:,.2f}")

,Cambio anual de población en 2020
Media,"2,124,908.85"
Mediana,"34,293.50"
Varianza,"90,687,335,031,103.64"
Desviación estándar,"9,522,989.82"
Mínimo,"-494,016.00"
Máximo,"75,707,590.00"


In [9]:
fila_minimo = df_2020.loc[cambio_2020.idxmin(), ["Entity", variable]]
fila_maximo = df_2020.loc[cambio_2020.idxmax(), ["Entity", variable]]

entidades_extremas = pd.DataFrame({
    "Mínimo": fila_minimo,
    "Máximo": fila_maximo,
}).T
entidades_extremas

,Entity,Annual change in population
Mínimo,Venezuela,-494016.0
Máximo,World,75707590.0


In [10]:
# Comprobación de la media y la mediana con NumPy
print("Media con NumPy:", np.mean(cambio_2020))
print("Mediana con NumPy:", np.median(cambio_2020))

Media con NumPy: 2124908.84765625
Mediana con NumPy: 34293.5


### Respuestas de la Actividad 3

1. **¿Cuál es el valor de la media?** La media es **2,124,908.85 habitantes**.
2. **¿Cuál es el valor de la mediana?** La mediana es **34,293.50 habitantes**.
3. **¿Cuál es el valor de la desviación estándar?** La desviación estándar muestral es **9,522,989.82 habitantes**.
4. **¿Qué entidad presenta el valor mínimo y cuál el máximo?** **Venezuela** presenta el mínimo, con **-494,016 habitantes**, y **World** presenta el máximo, con **75,707,590 habitantes**. `World` aparece porque el dataset incluye agregados además de países.

## Actividad 4 — Comprobación del entorno

**Propósito:** comprobar que el notebook está utilizando correctamente las herramientas configuradas para la práctica.

La siguiente celda muestra las versiones de Python, Pandas y NumPy del entorno virtual activo.

In [11]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.13.15 (tags/v3.13.15:4061bc4, Aug  5 2026, 13:05:39) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


In [12]:
df.loc[:, "Entity"]

0        Afghanistan
1        Afghanistan
2        Afghanistan
3        Afghanistan
4        Afghanistan
            ...     
38395       Zimbabwe
38396       Zimbabwe
38397       Zimbabwe
38398       Zimbabwe
38399       Zimbabwe
Name: Entity, Length: 38400, dtype: str

In [13]:
df.iloc[600:610,0:12]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
600,American Samoa,ASM,1951,380.0,NaN
601,American Samoa,ASM,1952,126.0,NaN
602,American Samoa,ASM,1953,115.0,NaN
603,American Samoa,ASM,1954,91.0,NaN
604,American Samoa,ASM,1955,95.0,NaN
605,American Samoa,ASM,1956,83.0,NaN
606,American Samoa,ASM,1957,23.0,NaN
607,American Samoa,ASM,1958,-33.0,NaN
608,American Samoa,ASM,1959,-19.0,NaN
609,American Samoa,ASM,1960,228.0,NaN


In [14]:
df.loc[600:601, ["Entity", "Annual change in population"]]

,Entity,Annual change in population
600,American Samoa,380.0
601,American Samoa,126.0


In [15]:
df.iloc[:, 0:2]

,Entity,Code
0,Afghanistan,AFG
1,Afghanistan,AFG
2,Afghanistan,AFG
3,Afghanistan,AFG
4,Afghanistan,AFG
...,...,...
38395,Zimbabwe,ZWE
38396,Zimbabwe,ZWE
38397,Zimbabwe,ZWE
38398,Zimbabwe,ZWE


In [16]:
df.iloc[:, 0:4]
 

 

,Entity,Code,Year,Annual change in population
0,Afghanistan,AFG,1951,103163.0
1,Afghanistan,AFG,1952,108441.0
2,Afghanistan,AFG,1953,108919.0
3,Afghanistan,AFG,1954,111251.0
4,Afghanistan,AFG,1955,119027.0
...,...,...,...,...
38395,Zimbabwe,ZWE,2096,NaN
38396,Zimbabwe,ZWE,2097,NaN
38397,Zimbabwe,ZWE,2098,NaN
38398,Zimbabwe,ZWE,2099,NaN


In [17]:
df.iloc[[0, 2, 3]]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN


In [18]:
df.iloc[:,[0,1,3]]

,Entity,Code,Annual change in population
0,Afghanistan,AFG,103163.0
1,Afghanistan,AFG,108441.0
2,Afghanistan,AFG,108919.0
3,Afghanistan,AFG,111251.0
4,Afghanistan,AFG,119027.0
...,...,...,...
38395,Zimbabwe,ZWE,NaN
38396,Zimbabwe,ZWE,NaN
38397,Zimbabwe,ZWE,NaN
38398,Zimbabwe,ZWE,NaN


In [19]:
df.loc[(df["Entity"] == "Mexico") & (df["Year"] >= 2000),
    ["Entity", "Year", "Annual change in population"]]

,Entity,Year,Annual change in population
22249,Mexico,2000,1509843.0
22250,Mexico,2001,1473548.0
22251,Mexico,2002,1449526.0
22252,Mexico,2003,1429890.0
22253,Mexico,2004,1415614.0
...,...,...,...
22345,Mexico,2096,NaN
22346,Mexico,2097,NaN
22347,Mexico,2098,NaN
22348,Mexico,2099,NaN


In [20]:
df.loc[(df["Entity"] == "Mexico") & (df["Year"] == 2000),
    ["Entity", "Year", "Annual change in population"]]

,Entity,Year,Annual change in population
22249,Mexico,2000,1509843.0


In [ ]:
df = df.set_index("Entity")

,Code,Year,Annual change in population,Annual population change (Projected)
Entity,,,,
Afghanistan,AFG,1951,103163.0,NaN
Afghanistan,AFG,1952,108441.0,NaN
Afghanistan,AFG,1953,108919.0,NaN
Afghanistan,AFG,1954,111251.0,NaN
Afghanistan,AFG,1955,119027.0,NaN
...,...,...,...,...
Zimbabwe,ZWE,2096,NaN,100752.0
Zimbabwe,ZWE,2097,NaN,91796.0
Zimbabwe,ZWE,2098,NaN,87605.0
